ArduPlane/soaring.cpp

```diff
@@ -66,156 +66,118 @@ void Plane::update_soaring() {
+                set_mode(*exit_mode, ModeReason::SOARING_ALT_TOO_HIGH);
                 break;
-            }
-            switch (loiterStatus) {
-                case SoaringController::LoiterStatus::ALT_TOO_HIGH:
-                    gcs().send_text(MAV_SEVERITY_INFO, "%s%s", strTooHigh, mode_auto.name());
-                    set_mode(mode_auto, ModeReason::SOARING_ALT_TOO_HIGH);
-                    break;
-                case SoaringController::LoiterStatus::ALT_TOO_LOW:
-                    gcs().send_text(MAV_SEVERITY_INFO, "%s%s", strTooLow, mode_auto.name());
-                    set_mode(mode_auto, ModeReason::SOARING_ALT_TOO_LOW);
-                    break;
-                default:
-                case SoaringController::LoiterStatus::THERMAL_WEAK:
-                        gcs().send_text(MAV_SEVERITY_INFO, "%s%s", strTooWeak, mode_auto.name());
-                        set_mode(mode_auto, ModeReason::SOARING_THERMAL_ESTIMATE_DETERIORATED);
-                    break;
-                case SoaringController::LoiterStatus::DRIFT_EXCEEDED:
-                    gcs().send_text(MAV_SEVERITY_INFO, "%s%s", strTooFar, mode_auto.name());
-                    set_mode(mode_auto, ModeReason::SOARING_DRIFT_EXCEEDED);
-                    break;
-                } // switch loiterStatus
-            break;
-            } // case AUTO
-        default: // all other modes
-            break;
-        } // switch previous_mode
+            case SoaringController::LoiterStatus::ALT_TOO_LOW:
+                gcs().send_text(MAV_SEVERITY_INFO, "Soaring: Too low, restoring %s", exit_mode->name());
+                set_mode(*exit_mode, ModeReason::SOARING_ALT_TOO_LOW);
+                break;
+            default:
+            case SoaringController::LoiterStatus::THERMAL_WEAK:
+                gcs().send_text(MAV_SEVERITY_INFO, "Soaring: Thermal ended, restoring %s", exit_mode->name());
+                set_mode(*exit_mode, ModeReason::SOARING_THERMAL_ESTIMATE_DETERIORATED);
+                break;
+            case SoaringController::LoiterStatus::DRIFT_EXCEEDED:
+                gcs().send_text(MAV_SEVERITY_INFO, "Soaring: Drifted too far, restoring %s", exit_mode->name());
+                set_mode(*exit_mode, ModeReason::SOARING_DRIFT_EXCEEDED);
+                break;
+        } // switch loiterStatus
+
         break;
-        } // case loiter
+       
+    } // case loiter
     } // switch control_mode
+}
 
 
-    if (control_mode == &mode_loiter) {
-        // still in thermal - need to update the wp location
-        g2.soaring_controller.get_target(next_WP_loc);
-
+bool Plane::soaring_exit_heading_aligned() const
+{
+    // Return true if the current heading is aligned with the next objective.
+    // If home is not set, or heading not locked, return true to avoid delaying mode change.
+    switch (previous_mode->mode_number()) {
+        case Mode::Number::AUTO: {
```

Comment:
The rest of this file doesn't have `case` indented from `switch`, and while I blieve soaring.cpp is your domain, I think tridge has a preference for indented-to-same-level.

Reply:
All fixed, thanks.